benchexec/tools/FuSeBMC.py

```diff
@@ -0,0 +1,79 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import benchexec.util as util
+import benchexec.tools.template
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool):
+    """
+    This class serves as tool adaptor for ESBMC (http://www.esbmc.org/)
+    """
+
+    REQUIRED_PATHS = ["esbmc", "esbmc-wrapper.py", "my_instrument"]
+
+    def executable(self):
+        return util.find_executable("esbmc-wrapper.py")
+
+    def working_directory(self, executable):
+        executableDir = os.path.dirname(executable)
+        return executableDir
+
+    def version(self, executable):
+        return self._version_from_tool(executable, "-v")
+
+    def name(self):
+        return "FuSeBMC"
+
+    def cmdline(self, executable, options, tasks, propertyfile, rlimits):
+        assert len(tasks) == 1, "only one inputfile supported"
```

Comment:
After changing to `BaseTool2`, this assert is no longer necessary because you can simply use `task.single_input_file`.